static/app/views/eventsV2/table/queryField.tsx

```diff
@@ -101,18 +102,32 @@ type OptionType = {
 
 class QueryField extends React.Component<Props> {
   FieldSelectComponents = {
-    Option: ({label, data, ...props}: OptionProps<OptionType>) => (
-      <components.Option label={label} data={data} {...props}>
-        <span data-test-id="label">{label}</span>
-        {this.renderTag(data.value.kind)}
-      </components.Option>
-    ),
-    SingleValue: ({data, ...props}: SingleValueProps<OptionType>) => (
-      <components.SingleValue data={data} {...props}>
-        <span data-test-id="label">{data.label}</span>
-        {this.renderTag(data.value.kind)}
-      </components.SingleValue>
-    ),
+    Option: ({label, data, ...props}: OptionProps<OptionType>) => {
+      if (data.value === undefined)
```

Comment:
```suggestion
      if (data.value === '' || data.value === undefined)
```

Reply:
@edwardgou-sentry You're missing some (preferred) braces here, and you can just use truthiness or the `defined` function.

Reply:
Totally agree truthiness is not completely covered here.